### ISSUE TITLE
How to calculate the deconvolution and BN layers？

### ISSUE THREAD

**Issue by Hans1984** [belltailjp/chainer_computational_cost#22]
Nice work, however, it seems that this tool doesn't support calculate the params and Flops of deconvolution and BN layers. Is there any solution?
Thanks a lot.

**Comment by belltailjp** [comment]
It does.
https://github.com/belltailjp/chainer_computational_cost/blob/master/DETAILS.md#deconvolution2dfunction
https://github.com/belltailjp/chainer_computational_cost/blob/master/DETAILS.md#fixedbatchnormalization
Or ain't they what you need?

**Comment by Hans1984** [comment]
Thanks a lot. I'm sorry, I didn't say it clearly. Since I need to calculate the 3D deconvolution. 
And the detail is as follow:
/usr/local/lib/python3.5/dist-packages/chainer_computational_cost/computational_cost_hook.py:251: UserWarning: chainer.functions.normalization.batch_normalization.BatchNormalization is not yet supported by ComputationalCostHook, ignored
  "ComputationalCostHook, ignored".format(fqn))
/usr/local/lib/python3.5/dist-packages/chainer_computational_cost/computational_cost_hook.py:251: UserWarning: chainer.functions.connection.deconvolution_nd.DeconvolutionND is not yet supported by ComputationalCostHook, ignored
  "ComputationalCostHook, ignored".format(fqn))
/usr/local/lib/python3.5/dist-packages/chainer_computational_cost/computational_cost_hook.py:251: UserWarning: chainer.functions.noise.dropout.Dropout is not yet supported by ComputationalCostHook, ignored
  "ComputationalCostHook, ignored".format(fqn))

**Comment by belltailjp** [comment]
For BN, `FixedBatchNormalization` (BN in validation mode) is supported so please use `chainer.using_config('train', False)` when running the forward pass, as shown in the README (See the Quickstart chapter).

For Convolution/DeconvolutionND, right, it's not supported yet, but I'm open to any contributions.
Here's the implementation of deconv2d, which is not that complicated and shouldn't be different from deconvNd that much, I guess.
https://github.com/belltailjp/chainer_computational_cost/blob/master/chainer_computational_cost/cost_calculators/connection.py#L131-L165

**Comment by Hans1984** [comment]
Thanks for your help.
